simulator/mips/t/unit_test.cpp

```diff
@@ -155,21 +139,30 @@ TEST( Func_instr_disasm, Process_Disasm)
 //  ASSERT_EQ(FuncInstr(0xe131fb2e).Dump(), "sc $s1, 0xfb2e($t1)");
     ASSERT_EQ(FuncInstr(0x1229ffff).Dump(), "beq $s1, $t1, -1");
     ASSERT_EQ(FuncInstr(0x1229000e).Dump(), "beq $s1, $t1, 14");
-//  ASSERT_EQ(FuncInstr(0x0621fffd).Dump(), "bgez $s1, 0xfffc");
-//  ASSERT_EQ(FuncInstr(0x0621000c).Dump(), "bgez $s1, 0xc");
-//  ASSERT_EQ(FuncInstr(0x0631fffb).Dump(), "bgezal $s1, 0xfffa");
-//  ASSERT_EQ(FuncInstr(0x0631000a).Dump(), "bgezal $s1, 0xa");
+    ASSERT_EQ(FuncInstr(0x0621fffc).Dump(), "bgez $s1, 0xfffc");
+    ASSERT_EQ(FuncInstr(0x0621000c).Dump(), "bgez $s1, 0xc");
+    ASSERT_EQ(FuncInstr(0x0631fffa).Dump(), "bgezal $s1, 0xfffa");
+    ASSERT_EQ(FuncInstr(0x0631000a).Dump(), "bgezal $s1, 0xa");
     ASSERT_EQ(FuncInstr(0x1e20fff9).Dump(), "bgtz $s1, -7");
     ASSERT_EQ(FuncInstr(0x1e200008).Dump(), "bgtz $s1, 8");
     ASSERT_EQ(FuncInstr(0x1a20fff7).Dump(), "blez $s1, -9");
     ASSERT_EQ(FuncInstr(0x1a200006).Dump(), "blez $s1, 6");
-//  ASSERT_EQ(FuncInstr(0x0630fff5).Dump(), "bltzal $s1, -11");
-//  ASSERT_EQ(FuncInstr(0x06300004).Dump(), "bltzal $s1, 4");
-//  ASSERT_EQ(FuncInstr(0x0620fff3).Dump(), "bltz $s1, -13");
-//  ASSERT_EQ(FuncInstr(0x06200002).Dump(), "bltz $s1, 2");
+    ASSERT_EQ(FuncInstr(0x0630fff5).Dump(), "bltzal $s1, 0xfff5");
+    ASSERT_EQ(FuncInstr(0x06300004).Dump(), "bltzal $s1, 0x4");
```

Comment:
There is an inconsistency. Old branches use signed decimal outputs, while new branches use unsigned hex. IMO signed decimal output is better, as you can understand where the target of branch is.

Reply:
I do it because i have problems with unit test, but same instr bgezal use hex.
How I should to rewrite it?

Reply:
Yes, it's OK to rewrite them.

Reply:
Rewrite bltzal and bgezal? That last arg will be dec?

Reply:
Yes. the immediate should be signed decimal number.